packages/flame/lib/src/components/text_component.dart

```diff
@@ -1,16 +1,30 @@
 import 'dart:ui';
 
 import 'package:flame/components.dart';
+import 'package:flame/src/text/formatter_text_renderer.dart';
+import 'package:flame/src/text/inline/text_element.dart';
 import 'package:flame/src/text/text_renderer.dart';
 import 'package:flutter/painting.dart';
 import 'package:meta/meta.dart';
 
 class TextComponent<T extends TextRenderer> extends PositionComponent {
-  String _text;
-  T _textRenderer;
+  TextComponent({
+    String? text,
+    T? textRenderer,
+    super.position,
+    super.size,
+    super.scale,
+    super.angle,
+    super.anchor,
+    super.children,
+    super.priority,
+  })  : _text = text ?? '',
+        _textRenderer = textRenderer ?? TextRenderer.createDefault<T>() {
+    updateBounds();
+  }
 
   String get text => _text;
-
+  String _text;
```

Comment:
```dart
  String _text;
  String get text => _text;
```

Normally we have the private at the top and the getter/setter below it right?

Reply:
Doesn't that style become weird when adding dartdocs to the getter though? 🤔

Reply:
As far as I can see, there is no Dart or Flutter official recommendations about this. The style that I'm currently using is taken from the example here: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#perform-dirty-checks-in-setters. 

In practice, Flutter repo uses several styles: either
```dart
  /// The parent of this node in the tree.
  AbstractNode? get parent => _parent;
  AbstractNode? _parent;

  /// The current value stored in this notifier.
  ///
  /// When the value is replaced with something that is not equal to the old
  /// value as evaluated by the equality operator ==, this class notifies its
  /// listeners.
  @override
  T get value => _value;
  T _value;
  set value(T newValue) {
    if (_value == newValue)
      return;
    _value = newValue;
    notifyListeners();
  }
```
or
```dart
  /// Current code point.
  ///
  /// If the iterator has hit either end, the [_currentCodePoint] is -1
  /// and `_position == _nextPosition`.
  int _currentCodePoint = -1;

  ...

  /// The rune (integer Unicode code point) starting at the current position in
  /// the string.
  ///
  /// The value is -1 if there is no current code point.
  int get current => _currentCodePoint;
```

I guess we just need to decide which style we like more, and put this into our own style guide.

Reply:
I like the first one, since we usually don't add dartdocs to both the public facing getter and the underlying value.